### ISSUE TITLE
Long processing time on 1.5_single_cell_statistics.py

### ISSUE THREAD

**Issue by asjureka** [scharch/zap#19]
Hi,

I was recently running the SONAR workflow on our HPC cluster, and I noticed that it was taking quite a long time to finish that step. When I went through the script, it appears that this particular script isn't threaded (or not obviously, please correct me if I'm wrong). Would it be possible to add threading to this script to help it utilize available processing power more efficiently? 

Thank you!

**Comment by scharch** [comment]
Yes, this is a known issue. The main cause is large 10x datasets with droplets containing 10s of (usually light) chains that trigger an inordinate [number of alignments trying to collapse them](https://github.com/scharch/SONAR/blob/74b5b9734f23ddceffd06b1e02ecd18e392c9567/annotate/1.5-single_cell_statistics.py#L142). Despite the function name, it is [actually extremely slow](https://github.com/scharch/SONAR/blob/011e0a6caebf15fc4f7aa826bbb2257264d48a0c/__init__.py#L345), writing to disk, calling muscle, and then reading the output.

I have some ideas about how to fix this, which I hope to include as part of a major overhaul/refactor of module 1 planned for the next major version of SONAR, but I don't have any sort of timeline for release yet.

In the meantime, you are welcome to submit a pull request to add threading, though I'm not sure how much of a speed up you can get that way. A better approach would probably be to filter your rearrangements TSV prior to calling to 1.5 to remove "cells" with more than 5-10 chains present --they are likely to be background noise/unsalvageable, anyway.
